src/core/qs.js

```diff
@@ -66,7 +66,9 @@ DM.provide('QS',
     decode: function(str)
     {
         var qsParams = str.split('&');
-        var decode = decodeURIComponent;
+        var decode = function(string) {
+            return decodeURIComponent(string.replace(/\+/g, ' '));
```

Comment:
The fundamental question is, is there any possibility to get strings with `+` here that are legit and we don't want them replaced with spaces and negatively affected?

There are 5 callers of `DM.QS.decode` in `player.js`, `cookie.js` and `auth.js`.

I'm most concerned about the detection of `apimode=json`, I guess once we assume JSON mode, the + to space conversion should not occur:

```

    _decodePostMessage: function(rawMessage)
    {
      if (rawMessage.substring(0, 1) === '{') {
        try {
          var data = JSON.parse(rawMessage);
          return data;
        }
        catch(e) {
          return {};
        }
      }
      return DM.QS.decode(rawMessage);
    },
```

Reply:
I think we're safe 😬